CMakeLists.txt

```diff
@@ -261,9 +261,8 @@ set (nlopt_lib nlopt)
 add_library (${nlopt_lib} ${NLOPT_SOURCES})
 add_dependencies(${nlopt_lib} generate-cpp)
 if(NLOPT_WITH_LUKSAN)
-  set(nlopt_lib_compile_definitions WITH_LUKSAN)
+  target_compile_definitions(${nlopt_lib} PUBLIC NLOPT_WITH_LUKSAN)
```

Comment:
it must be private (since its only used in optimize.c)

Reply:
good catch